### PR TITLE
OBPIH-2595: Improve importing order items from template

### PR DESCRIPTION
**Pull request by awalkowiak**

### CODE REVIEW

File: grails-app/services/org/pih/warehouse/order/OrderService.groovy [hunk 1]
```diff
@@ -593,12 +604,34 @@ class OrderService {
                         BigDecimal qtyPerUom = uomParts.length > 1 ? BigDecimal.valueOf(Double.valueOf(uomParts[1])) : null
                         orderItem.quantityUom = uom
                         orderItem.quantityPerUom = qtyPerUom
+                    } else {
+                        throw new IllegalArgumentException("Missing unit of measure.")
```
Comment (jmiranda) [inline grails-app/services/org/pih/warehouse/order/OrderService.groovy:608]: We could potentially default to EA, but I think it's better to err on the side of caution for now.

File: grails-app/services/org/pih/warehouse/order/OrderService.groovy [hunk 2]
```diff
@@ -593,12 +604,34 @@ class OrderService {
...
+                    }
+
+                    Integer parsedQty = Integer.valueOf(quantity)
+                    if (parsedQty <= 0) {
+                        throw new IllegalArgumentException("Wrong quantity value: ${parsedQty}.")
+                    }
+
+                    BigDecimal parsedUnitPrice = new BigDecimal(unitPrice).setScale(2, RoundingMode.FLOOR)
```
Comment (jmiranda) [inline grails-app/services/org/pih/warehouse/order/OrderService.groovy:616]: We actually want this to be 4 places. We should ask Kelsey what she expects the rounding mode to be.

Comment (jmiranda) [reply grails-app/services/org/pih/warehouse/order/OrderService.groovy:616]: Actually, sorry I'm probably wrong about the 4 places. That's only really needed for unit price on product.

Comment (awalkowiak) [reply grails-app/services/org/pih/warehouse/order/OrderService.groovy:616]: In the spreadsheet with the column description, Kelsey requested to limit to two decimal places when parsing values.

**Pull request merged**